presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusClient.java

```diff
@@ -0,0 +1,192 @@
+            throws URISyntaxException
+    {
+        requireNonNull(config, "config is null");
+        requireNonNull(metricCodec, "metricCodec is null");
+        requireNonNull(typeManager, "typeManager is null");
+
+        tables = Suppliers.memoize(metricsSupplier(metricCodec, getPrometheusMetricsURI(config)));
+        prometheusURI = config.getPrometheusURI();
+        queryChunkSizeDuration = config.getQueryChunkSizeDuration();
+        this.typeManager = typeManager;
+    }
+
+    public Set<String> getSchemaNames()
+    {
+        return ImmutableSet.of("prometheus");
+    }
+
+    private URI getPrometheusMetricsURI(PrometheusConfig config)
+            throws URISyntaxException
+    {
+        // endpoint to retrieve metric names from Prometheus
+        return new URI(config.getPrometheusURI().toString().concat(METRICS_ENDPOINT));
+    }
+
+    public Set<String> getTableNames(String schema)
+    {
+        requireNonNull(schema, "schema is null");
+        if (schema.equals("prometheus")) {
+            String status = (String) tables.get().get("status");
+            if (status.equals("success")) {
+                List<String> tableNames = (List<String>) tables.get().get("data");
+                if (tableNames == null) {
+                    return ImmutableSet.of();
+                }
+                return tableNames.stream().collect(Collectors.toSet());
+            }
+            else {
+                //TODO this deals with success|error, there may be warnings to handle
+                return ImmutableSet.of();
+            }
+        }
+        return ImmutableSet.of();
+    }
+
+    public PrometheusTable getTable(String schema, String tableName)
+    {
+        requireNonNull(schema, "schema is null");
+        requireNonNull(tableName, "tableName is null");
+        if (schema.equals("prometheus")) {
+            List<String> tableNames = (List<String>) tables.get().get("data");
+            if (tableNames == null) {
+                return null;
+            }
+            if (!tableNames.contains(tableName)) {
+                return null;
+            }
+            PrometheusTable table = new PrometheusTable(
+                    tableName,
+                    ImmutableList.of(
+                            new PrometheusColumn("labels", typeManager.getType(TypeSignature.parseTypeSignature("map(varchar, varchar)"))),
+                            new PrometheusColumn("timestamp", TimestampType.TIMESTAMP),
+                            new PrometheusColumn("value", DoubleType.DOUBLE)),
+                    ImmutableList.of(prometheusURI));
+            return table;
+        }
+        else {
+            return null;
+        }
+    }
+
+    private static Supplier<Map<String, Map<String, PrometheusTable>>> schemasSupplier(final JsonCodec<Map<String, List<PrometheusTable>>> catalogCodec, final URI metadataUri)
```

Comment:
Is this used? I don't think we have more than 1 schema anyways.

Reply:
Addressed in 80f9ba51